.github/workflows/node.js.yml

```diff
@@ -6,19 +6,58 @@ name: Node.js CI
 on: [push, pull_request]
 
 jobs:
-  build:
-
+  common:
+    name: Common checks (Node.js v${{ matrix.node-version }})
     runs-on: macos-latest
-
     strategy:
       matrix:
         node-version: [12.x, 14.x, 16.x]
+    steps:
+      - uses: actions/checkout@v2
+      - uses: bahmutov/npm-install@v1
+        with:
+          useRollingCache: true
+      - run: npm run test:common
 
+  ios-simulator:
+    name: ios-simulator checks (Node.js v${{ matrix.node-version }} / iOS v${{ matrix.ios-version }} / Xcode v${{ matrix.xcode-version }} / ${{ matrix.device-name }})
+    runs-on: macos-latest
+    strategy:
+      matrix:
+        node-version: [12.x, 14.x, 16.x]
```

Comment:
I'd rather prefer to use the recent node version and different xcode versions in the matrix here.